### PR TITLE
[lldb] Avoid repeated map lookups (NFC)

### PR DESCRIPTION
**Pull request by kazutakahirata**

### CODE REVIEW

File: lldb/source/Plugins/Process/gdb-remote/ProcessGDBRemote.cpp [hunk 1]
```diff
@@ -5324,7 +5324,7 @@ std::string ProcessGDBRemote::HarmonizeThreadIdsForProfileData(
         std::map<uint64_t, uint32_t>::iterator iterator =
             m_thread_id_to_used_usec_map.find(thread_id);
         if (iterator != m_thread_id_to_used_usec_map.end()) {
-          prev_used_usec = m_thread_id_to_used_usec_map[thread_id];
+          prev_used_usec = iterator->second;
         }
```
Comment (JDevlieghere) [inline lldb/source/Plugins/Process/gdb-remote/ProcessGDBRemote.cpp:5328]: While you're here, might as well remove the braces around the single-line `if`.

**Review by JDevlieghere — APPROVED** [review]
LGTM

File: lldb/source/Plugins/Process/gdb-remote/ProcessGDBRemote.cpp (hunk 1 above)
Comment (kazutakahirata) [reply lldb/source/Plugins/Process/gdb-remote/ProcessGDBRemote.cpp]: Sure.  Addressed in the latest iteration.

**Comment by jimingham** [comment]
Note that a lot of lldb was written before C++11 was allowed in llvm projects.  So looking through code that hasn't been touched recently, you will likely find constructs that could be made more efficient with some of the newer C++ gadgets.  We probably shouldn't be doing change just for changes sake, but when we can make the code clearer or more efficient, that's certainly worth doing.

Jim

**Pull request merged**